### ISSUE TITLE
[✨] Add way to specify PLATFORM for action$

### ISSUE THREAD

**Issue by mewhhaha** [BuilderIO/qwik#2952]
### Is your feature request related to a problem?

Love using qwik! I hope I'm doing this correctly.

```tsx
export interface ActionConstructor {
  <O>(actionQrl: (form: JSONObject, event: RequestEventAction) => ValueOrPromise<O>): Action<O>;
  <O, B extends ZodReturn>(
    actionQrl: (data: GetValidatorType<B>, event: RequestEventAction) => ValueOrPromise<O>,
    options: B
  ): Action<
    O | FailReturn<z.typeToFlattenedError<GetValidatorType<B>>>,
    GetValidatorType<B>,
    false
  >;
}
```

In the type signature for `action$` the `PLATFORM` for `RequestEventAction` defaults to unknown, unlike in `loader$` where you can pass a generic type.

### Describe the solution you'd like

Add a generic type to `action$` that gets passed to `RequestEventAction` so that you can specify the `PLATFORM`. The same way that is possible to do it for `loader$`.  
  
OR  
  
Similar to Remix have both `loader$` and `action$` use an empty interface that is exposed for `PLATFORM` instead of a generic type. This would make it so that you can make one declaration app-wide. (link to Remix MR that did something similar)[https://github.com/remix-run/remix/pull/1876 ].

In Remix they have `AppLoadContext` that is their equivalent of `PLATFORM`. By including a declaration file like below you can merge it and have `AppLoadContext` be whatever you want in the application. So you don't have to type `AppLoadContext` at each instance when you use a loader or an action.

```
declare module "@remix-run/cloudflare" {
  interface AppLoadContext {
    ...
  }
}
```

### Describe alternatives you've considered

* Casting `platform` to the proper type.
* `@ts-ignore` the error
* Using a patching tool like pnpm patch

**Comment by mewhhaha** [comment]
I tried out `qwik-city 0.2.1` which seems to have resolved this issue with the updated definition. 🎉 Thank you!

```tsx
export declare interface RequestEventAction<PLATFORM = QwikCityPlatform> extends RequestEventCommon<PLATFORM> {
    fail: <T extends Record<string, any>>(status: number, returnData: T) => FailReturn<T>;
}
```

Given the definition above I added the below to a declaration file and I was able to get typings for `platform` in `action$`

```tsx
import "@builder.io/qwik-city/middleware/request-handler";

declare module "@builder.io/qwik-city/middleware/request-handler" {
  interface QwikCityPlatform {
    foobar: string;
  }
}
```
<img width="558" alt="Screenshot 2023-02-16 at 00 58 14" src="https://user-images.githubusercontent.com/3399205/219222656-abcb7ddb-f7cb-4e5d-a099-831a2fbc2b6d.png">

However it seems that `loader$` is not having this properly applied to it. `loader$` uses `RequestEventLoader_2` which defaults to `QwikCityPlatform` ❤️  however 😨 the generic for `RequestEventLoader_2` is being set by the `loader$` function which defaults it to `unknown`, so even if `QwikCityPlatform` is defined it ends up being `unknown`.

```tsx
export declare const loader$: <RETURN, PLATFORM = unknown>(first: (event: RequestEventLoader_2<PLATFORM>) => RETURN) => Loader<RETURN>;
```

<img width="452" alt="Screenshot 2023-02-16 at 00 58 23" src="https://user-images.githubusercontent.com/3399205/219222711-be5d62ba-5fa3-4e8d-a2f2-4c13c805c0fd.png">
